python-sdk/src/astro/databases/snowflake.py

```diff
@@ -579,17 +566,26 @@ def load_pandas_dataframe_to_table(
         :param if_exists: Strategy to be used in case the target table already exists.
         :param chunk_size: Specify the number of rows in each batch to be written at a time.
         """
-        if if_exists == "replace":
-            self.create_table(target_table, dataframe=source_dataframe)
+
+        auto_create_table = False
+        if if_exists == "replace" or not self.table_exists(target_table):
+            auto_create_table = True
+
+        if target_table.metadata.schema:
+            target_table.metadata.schema = target_table.metadata.schema.upper()
+
+        if target_table.metadata.database:
+            target_table.metadata.database = target_table.metadata.database.upper()
 
         pandas_tools.write_pandas(
             conn=self.hook.get_conn(),
             df=source_dataframe,
-            table_name=target_table.name,
+            table_name=target_table.name.upper(),
```

Comment:
What are the implications of making the table name only uppercase here?

Reply:
1. This is the current behavior.
2.  Also, pandas_tools.write_pandas() is not able to create a table when a lowercase name is passed. However, it is possible in Snowflake to create a table in lowercase. Which will break if someone tries to point to an existing table which is in lowercase.